rules/slack_rules/slack_app_access_expanded.py

```diff
@@ -9,6 +9,10 @@
 
 
 def rule(event):
+    # if the action is app_scopes_expanded, check the scopes_added in p_alert_context not to be empty
+    if event.get("action") == "app_scopes_expanded":
+        if event.deep_get("p_alert_context", "scopes_added", default=[]) == []:
+            return False
     return event.get("action") in ACCESS_EXPANDED_ACTIONS
```

Comment:
```suggestion
if event.get("action") not in ACCESS_EXPANDED_ACTIONS:
        return False

    # Check to confirm that app scopes actually expanded or not
    if event.get("action") == "app_scopes_expanded":
        changes = get_scope_changes(event)
        if not changes["added"]:
            return False

    return True
```
The `p_alert_context` field doesn't exist on an event until after the rule runs, so we can't reference it in the rule logic. We'll need to calculate the scopes separately. (I offloaded that logic to a function since we're going to be duplicating it in `rule` and also in `alert_context`.